app/src/main/java/edu/wpi/first/shuffleboard/app/Shuffleboard.java

```diff
@@ -34,7 +37,32 @@ public void init() throws AlreadyLockedException, IOException {
       throw alreadyLockedException;
     }
 
-    NetworkTablesJNI.startClient("localhost", 1735);
+    NetworkTable.setClientMode();
+
+    ChangeListener<String> serverChangeListener = (observable, oldValue, newValue) -> {
+      if (newValue.matches("[1-9](\\d{1,3})?")) {
+        NetworkTable.setTeam(Integer.parseInt(newValue));
+      } else if (newValue.isEmpty()) {
+        NetworkTable.setIPAddress("localhost");
+      } else {
+        NetworkTable.setIPAddress(newValue);
```

Comment:
I'm not sure i'm happy with the undiscoverable behavior here. I think ideally it would have a "team number" prompt and a hidden "hostname" field that you could show by clicking on a link, something like "Or, set the server directly...").